app/assets/javascripts/discourse/app/components/user-menu/messages-list.js

```diff
@@ -77,13 +77,16 @@ export default class UserMenuMessagesList extends UserMenuNotificationsList {
     const topics = data.topics.map((t) => this.store.createRecord("topic", t));
     await Topic.applyTransformations(topics);
 
-    if (this.siteSettings.show_user_menu_avatars) {
+    if (
+      this.siteSettings.show_user_menu_avatars ||
+      !this.siteSettings.prioritize_username_in_ux
+    ) {
       // Populate avatar_template for lastPoster
       const usersById = new Map(data.users.map((u) => [u.id, u]));
       topics.forEach((t) => {
-        t.last_poster_avatar_template = usersById.get(
-          t.lastPoster.user_id
-        )?.avatar_template;
+        const lastPoster = usersById.get(t.lastPoster.user_id);
+        t.last_poster_avatar_template = lastPoster?.avatar_template;
+        t.last_poster_name = lastPoster?.name;
```

Comment:
Should this be set unconditionally? With the code change right now, the poster's name will be the user's name when `show_user_menu_avatars ` is set to `true` even though `prioritize_username_in_ux ` is set to `true` as well.